test/controllers/userJourney/EuGoodsControllerSpec.scala

```diff
@@ -0,0 +1,153 @@
+          status(result) mustBe OK
+          contentType(result) mustBe Some("text/html")
+          charset(result) mustBe Some("utf-8")
+          contentAsString(result) must include("Do you import or export goods from or to countries outside the EU?")
+      }
+    }
+  }
+
+  "return HTML when there's nothing in S4L and vatScheme contains no data" in {
+    when(mockS4LService.fetchAndGet[EuGoods]()
+      (Matchers.eq(S4LKey[EuGoods]), Matchers.any(), Matchers.any()))
+      .thenReturn(Future.successful(None))
+
+    when(mockVatRegistrationService.getVatScheme()(Matchers.any[HeaderCarrier]()))
+      .thenReturn(Future.successful(emptyVatScheme))
+
+    callAuthorised(EuGoodsController.show, mockAuthConnector) {
+      result =>
+        status(result) mustBe OK
+        contentType(result) mustBe Some("text/html")
+        charset(result) mustBe Some("utf-8")
+        contentAsString(result) must include("Do you import or export goods from or to countries outside the EU?")
+    }
+  }
+
+  s"POST ${vatTradingDetails.vatEuTrading.routes.EuGoodsController.show()} with Empty data" should {
+
+    "return 400" in {
+      AuthBuilder.submitWithAuthorisedUser(EuGoodsController.submit(), mockAuthConnector, fakeRequest.withFormUrlEncodedBody(
+      )) {
+        result =>
+          status(result) mustBe Status.BAD_REQUEST
+      }
+
+    }
+  }
+
+  s"POST ${vatTradingDetails.vatEuTrading.routes.EuGoodsController.submit()} with Eu Goods Yes selected" should {
+
+    "return 303" in {
+      val returnCacheMapEuGoods = CacheMap("", Map("" -> Json.toJson(EuGoods(EuGoods.EU_GOODS_YES))))
+
+      when(mockS4LService.saveForm[EuGoods]
+        (Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any()))
+        .thenReturn(Future.successful(returnCacheMapEuGoods))
+
+      AuthBuilder.submitWithAuthorisedUser(EuGoodsController.submit(), mockAuthConnector, fakeRequest.withFormUrlEncodedBody(
+        "euGoodsRadio" -> EuGoods.EU_GOODS_YES
+      )) {
+        response =>
+          status(response) mustBe Status.SEE_OTHER
+          redirectLocation(response).getOrElse("") mustBe s"${contextRoot}/business-activity-description"
+      }
+
+    }
+  }
+
+  s"POST ${vatTradingDetails.vatEuTrading.routes.EuGoodsController.submit()} with Eu Goods No selected" should {
+
+    "return 303" in {
+      val returnCacheMapEuGoods = CacheMap("", Map("" -> Json.toJson(EuGoods(EuGoods.EU_GOODS_NO))))
+
+      when(mockS4LService.saveForm[EuGoods]
+        (Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any()))
+        .thenReturn(Future.successful(returnCacheMapEuGoods))
+
+      AuthBuilder.submitWithAuthorisedUser(EuGoodsController.submit(), mockAuthConnector, fakeRequest.withFormUrlEncodedBody(
+        "euGoodsRadio" -> EuGoods.EU_GOODS_NO
+      )) {
+        response =>
+          status(response) mustBe Status.SEE_OTHER
+          redirectLocation(response).getOrElse("") mustBe s"${contextRoot}/business-activity-description"
```

Comment:
same here. i think no point checking for status code and a redirect location. 
one more thing, unless it's a complex expression, there's not need for curly braces when using string interpolation.

`s"$contextRoot/business-activity-description"`